datafusion/src/physical_plan/expressions/average.rs

```diff
@@ -102,7 +114,14 @@ impl AggregateExpr for Avg {
     }
 
     fn field(&self) -> Result<Field> {
-        Ok(Field::new(&self.name, DataType::Float64, true))
+        Ok(Field::new(&self.name, self.data_type.clone(), true))
+    }
+
+    fn create_accumulator(&self) -> Result<Box<dyn Accumulator>> {
+        Ok(Box::new(AvgAccumulator::try_new(
+            // avg is f64 or decimal
+            &self.data_type,
```

Comment:
if a sum of `decimal(10,2)` can be `decimal(20,2)` shouldn't the accumulator state also be `decimal(20,2)` to avoid overflow?

I think handling overflow is probably fine to for a later date / PR, but it is strange to me that there is a discrepancy between the type for `sum` and the accumulator type for computing `avg`

Reply:
The result type of phy expr(sum/avg) is same with each Accumulator, and it was decided by `sum_return_type` and `avg_return_type`.

If the column is decimal(8,2), the avg of this column must be less than `10^10-1`, but we need more digits to represent the decimal part. For example, The avg of `3,4,6` is `4.333333.....`, we should increase the scale part.

Reply:
add issue to track the overflow.
https://github.com/apache/arrow-datafusion/issues/1460